File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.js
/**
 * @param {string[]} bank
 * @return {number}
 */
var numberOfBeams = function(bank) {
    let prevCount = 0, ans = 0;
    for(let floor of bank) {
        let count = 0;
        for(let device of floor) {
            if(device === '1') count++;
        }
        ans += count * prevCount;
        if(count != 0) prevCount = count;
    } 
    return ans;
};

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int prevCount = 0, n = bank.length;
        int ans = 0;
        for(int i=0; i<n; i++) {
            int count = 0;
            for(int j=0; j<bank[i].length(); j++) {
                if(bank[i].charAt(j) == '1') count++;
            }
            ans += count * prevCount;
            if(count != 0) prevCount = count;
        }
        return ans;
    }
}